backend/tilavarauspalvelu/models/reservation/actions.py

```diff
@@ -240,33 +240,20 @@ def get_state_on_reservation_confirmed(self, payment_type: PaymentType | None) -
     def get_required_fields(
         self,
         *,
-        reservee_type: CustomerTypeChoice | None = None,
-        reservee_is_unregistered_association: bool | None = None,
+        reservee_type: ReserveeType | None = None,
     ) -> list[str]:
         if reservee_type is None:
             reservee_type = self.reservation.reservee_type
-        if reservee_is_unregistered_association is None:
-            reservee_is_unregistered_association = self.reservation.reservee_is_unregistered_association
 
         qs = ReservationMetadataField.objects.filter(
             metadata_sets_required__reservation_units__reservations=self.reservation,
         )
 
         # Some fields are never mandatory for an individual reserver even if marked so in the metadata.
-        if reservee_type == CustomerTypeChoice.INDIVIDUAL:
-            qs = qs.exclude(field_name__in=["home_city", "reservee_id", "reservee_organisation_name"])
+        if reservee_type == ReserveeType.INDIVIDUAL:
+            qs = qs.exclude(field_name__in=["reservee_identifier", "reservee_organisation_name"])
 
-        # `reservee_id` is not mandatory for unregistered associations.
-        if reservee_is_unregistered_association:
-            qs = qs.exclude(field_name__in=["reservee_id"])
```

Comment:
Shouldn't `reservee_id` still be removed from required fields if `type == NONPROFIT`?

Reply:
Yes, fixed.